src/components/Tabs/index.tsx

```diff
@@ -0,0 +1,173 @@
+/* eslint-disable react-hooks/rules-of-hooks */
+import React, {FC, ReactNode, useEffect, useRef, useState} from 'react';
+import {StyleSheet, View, ViewStyle, ScrollView, FlatList} from 'react-native';
+import {moderateScale, scaledForDevice} from '../../scale';
+import {base, grey, primary} from '../../theme/palette';
+import BaseButton from '../BaseButton';
+import Text from '../Text';
+import {viewportWidth} from '../../scale';
+import {isObject} from '../../utils';
+
+interface Scene {
+	title: string;
+	scene: ReactNode;
+	disabled?: boolean;
+}
+
+export const positions = {
+	top: 'top',
+	bottom: 'bottom',
+};
+
+export type positionsType = typeof positions;
+export type keyPosition = keyof positionsType;
+
+interface TabsProps {
+	scenes: Scene[];
+	initialTab?: number;
+	position?: keyPosition;
+	onPressTabCb?: (activeTab: number) => void;
+	scrollContentStyle?: ViewStyle;
+	style?: ViewStyle;
+}
+
+interface TitleTabProps {
+	title: string;
+	index: number;
+	disabled: boolean | undefined;
+}
+
+const Tabs: FC<TabsProps> = ({
+	scenes,
+	initialTab = 0,
+	position = positions.top,
+	onPressTabCb = () => {},
+	scrollContentStyle = {},
+	style = {},
+	...props
+}) => {
+	if (!scenes || !Array.isArray(scenes) || !scenes.length) {
+		return null;
+	}
+
+	const [activeTab, setActiveTab] = useState(initialTab);
```

Comment:
@dam788 los ajustes de estilos quedaron bien, revisate que, cuando le paso un valor a initialTab que no sea 0, me rompe con este error:

`Invariant Violation: scrollToIndex should be used in conjunction with getItemLayout or onScrollToIndexFailed, otherwise there is no way to know the location of offscreen indices or handle failures.`

Cualquier cosa, si vos haces la prueba y no te pasa avisame y lo vemos a ver si estoy bardeando con algo.

Reply:
si gonza, me encontre con unos problemas que no habia tenido en cuenta. Ahi los resolvi, gracias!